hugolib/page.go

```diff
@@ -663,6 +664,13 @@ func (p *Page) update(f interface{}) error {
 		p.Draft = !*published
 	}
 
+	if p.Date.IsZero() {
+		fi, err := os.Stat(filepath.Join(helpers.AbsPathify(viper.GetString("ContentDir")), p.File.Path()))
```

Comment:
You need to use the https://github.com/spf13/hugo/blob/master/hugofs/fs.go#L30 filesystem.